### PR TITLE
Sign in with accounts

### PR DESCRIPTION
**Pull request by nabil-dbz**
This pull request add to the project the option to sign in with Google, Facebook or Github account.

### CODE REVIEW

File: planet/src/main/webapp/js/authentication.js [hunk 1]
```diff
@@ -74,3 +74,67 @@ function signIn() {
         // ...
     });
 }
+
+function signInWithGoogle() {
+    const provider = new firebase.auth.GoogleAuthProvider();
```
Comment (emilyzhaoo) [inline planet/src/main/webapp/js/authentication.js]: Maybe try renaming these consts to be more specific, const should always be in capital case ex: GOOGLE_PROVIDER

File: planet/src/main/webapp/js/authentication.js [hunk 2]
```diff
@@ -74,3 +74,67 @@ function signIn() {
...
+
+function signInWithGithub() {
+    const provider = new firebase.auth.GithubAuthProvider();
+    signInWithProvider(provider);
+}
+
+function signInWithProvider(provider) {
+    firebase.auth().signInWithPopup(provider).then(function(result) {
+        // The signed-in user info.
+        var user = result.user;
```
Comment (emilyzhaoo) [inline planet/src/main/webapp/js/authentication.js]: nit: Use let instead of var

**Review by emilyzhaoo — APPROVED** [review]

File: planet/src/main/webapp/js/authentication.js (hunk 1 above)
Comment (nabil-dbz) [reply planet/src/main/webapp/js/authentication.js]: Fixed!

File: planet/src/main/webapp/js/authentication.js (hunk 2 above)
Comment (nabil-dbz) [reply planet/src/main/webapp/js/authentication.js]: Fixed!

**Review by azhang72 — APPROVED** [review]

**Pull request merged**